scripts/lint.py

```diff
@@ -21,19 +21,31 @@ def run(*args):
 def eprint(*args):
     print(*args, file=sys.stderr)
 
-if shutil.which(f"clang-format-{REQUIRED_CLANG_FORMAT_MAJOR}"):
-    clang_format = f"clang-format-{REQUIRED_CLANG_FORMAT_MAJOR}"
-elif shutil.which("clang-format"):
-    clang_format = "clang-format"
-    clang_format_version = run(clang_format, "--version").rstrip().split(" ")[2]
-    clang_format_major = int(clang_format_version.split(".")[0])
-    if clang_format_major != REQUIRED_CLANG_FORMAT_MAJOR:
-        eprint(f"warning: clang-format is wrong version (wanted {REQUIRED_CLANG_FORMAT_MAJOR}, got {clang_format_version})")
-else:
+def find_clang_format_binary_in_path():
+    explicit_version = f"clang-format-{REQUIRED_CLANG_FORMAT_MAJOR}"
+    if shutil.which(explicit_version):
+        return explicit_version
+
+    implicit_version = "clang-format"
+    if shutil.which(implicit_version):
+        return implicit_version
+
+    # Can't find any version of clang-format in PATH.
+    return None
+
+clang_format = find_clang_format_binary_in_path()
+if not clang_format:
     eprint("error: clang-format is not installed.")
     eprint(f"Install clang-format version {REQUIRED_CLANG_FORMAT_MAJOR}")
     exit(1)
 
+clang_format_version = run(clang_format, "--version").rstrip().split(" ")[2]
+eprint('checking C/C++ formatting with clang-format %s' % clang_format_version)
```

Comment:
This is the first non-error output we're introducing in the script, so I wasn't sure if we wanted it to be to stdout or stderr.

Reply:
Stderr: there are non-error outputs below and they all pertain to actual lint diffs. Stderr is good for log messages. I also hid this behind the new `-v` verbose flag.